File: tests/visualization/mountainwave.py
import subprocess
import numpy as np
import matplotlib.pyplot as plt
import tools
import style

# Get host and user name.
host_name = subprocess.getoutput("hostname")
user_name = subprocess.getoutput("whoami")

if "levante" in host_name:
  # Levante cluster
  data_path = "/scratch/b/" + user_name + "/PF/runs"
  reference_path = "/scratch/b/" + user_name + "/PF/pinc/reference"

elif "login" in host_name:
  # Goethe cluster
  data_path = "/scratch/atmodynamics/" + user_name + "/PF/runs"
  reference_path = "/scratch/atmodynamics/" + user_name + "/PF/pinc/reference"

else:
  # Local machine
  data_path = "../"
  reference_path = "../"

print("data_path =", data_path)
print("reference_path =", reference_path)

# Import data.
data = tools.ModelOutput(data_path + 'mountainwave/')
reference = tools.ModelOutput(reference_path + 'mountainwave/')

# Set slices.
it = - 1

print(" ".join(("Time:", str(data.variables['t'][it]), "s")))

# Set grid.
xx = data.variables['x'][:] * 0.001 - 30
zz = data.variables['z'][it, :, 0] * 0.001
hh = data.variables['hm'][it, 0] * 0.001
xx = xx * np.ones_like(zz)

# Get vertical wind.
ww = data.groups['atmvar'].variables['w'][it, :, 0]
wref = reference.groups['atmvar'].variables['w'][it, :, 0]
deltaw = ww - wref

# Create plot.
(levels, colormap) = style.symmetric_contours(ww.min(), ww.max())
(fig, axes) = plt.subplots()
plot = axes.contourf(xx, zz, ww, levels, cmap = colormap)
axes.plot(xx[0], zz[0], color = "black", linewidth = 1.0)
axes.set_xlim(- 10.0, 10.0)
axes.set_ylim(0.0, 10.0)
axes.set_xlabel(r"$x \, \mathrm{\left[km\right]}$")
axes.set_ylabel(r"$z \, \mathrm{\left[km\right]}$")
fig.colorbar(plot, label = r"$w \, \mathrm{\left[m \, s^{- 1}\right]}$")
fig.savefig("".join((data_path, "/results/mountainwave.png")), dpi = 500)

# Create difference plot.
if data_path != reference_path:
  (levels, colormap) = style.symmetric_contours(deltaw.min(), deltaw.max())
  (fig, axes) = plt.subplots()
  plot = axes.contourf(xx, zz, deltaw, levels, cmap = colormap)
  axes.plot(xx[0], zz[0], color = "black", linewidth = 1.0)
  axes.set_xlim(- 10.0, 10.0)
  axes.set_ylim(0.0, 10.0)
  axes.set_xlabel(r"$x \, \mathrm{\left[km\right]}$")
  axes.set_ylabel(r"$z \, \mathrm{\left[km\right]}$")
  fig.colorbar(plot, label = r"$\Delta w \, \mathrm{\left[m \," \
      r"s^{- 1}\right]}$")
  fig.savefig("".join((data_path, "/results/mountainwave_difference.png")), \
      dpi = 500)

File: tests/visualization/mountainwave.jl
include("style.jl")
include("tools.jl")

using LaTeXStrings

# Set paths.
data_path = "../"
reference_path = "../"

println("data_path =", data_path)
println("reference_path =", reference_path)

# Import data.
data = ModelOutput(data_path * "mountainwave")
reference = ModelOutput(reference_path * "mountainwave")

println("Time: ", data.variables["t"][end], " s")

# Set grid.
x = data.variables["x"][99:202] .* 0.001 .- 30
z = data.variables["z"][99:202, 1, 1:51, end] .* 0.001
x = x .* ones(size(z))

# Get vertical wind.
w = data.groups["atmvar"].variables["w"][99:202, 1, 1:51, end]
wr = reference.groups["atmvar"].variables["w"][99:202, 1, 1:51, end]
deltaw = w .- wr

# Create plot.
(levels, colormap) = symmetric_contours(minimum(w), maximum(w))
contours = contourf(x, z, w; levels = levels, cmap = colormap)
plot(x[:, 1], z[:, 1]; color = "black", linewidth = 1.0)
xlim(-10.0, 10.0)
ylim(0.0, 10.0)
xlabel(L"x\,\left[\mathrm{km}\right]")
ylabel(L"z\,\left[\mathrm{km}\right]")
colorbar(contours; label = L"w\,\left[\mathrm{m\,s^{-1}}\right]")
savefig(data_path * "/results/mountainwave.png")

# Create difference plot.
if data_path != reference_path
  (levels, colormap) = symmetric_contours(minimum(deltaw), maximum(deltaw))
  clf()
  contours = contourf(x, z, deltaw; levels = levels, cmap = colormap)
  plot(x[:, 1], z[:, 1]; color = "black", linewidth = 1.0)
  xlim(-10.0, 10.0)
  ylim(0.0, 10.0)
  xlabel(L"x\,\left[\mathrm{km}\right]")
  ylabel(L"z\,\left[\mathrm{km}\right]")
  colorbar(contours; label = L"\Delta w\,\left[\mathrm{m\,s^{-1}}\right]")
  savefig(data_path * "/results/mountainwave_difference.png")
end
